### PR TITLE
Compiletest

### PR DESCRIPTION
**Pull request by Redecorating**
Test applying the patches and compiling the kernel with only modules for t2 hardware enabled.

Use a file called `version` to determine which kernel version to apply the patches to.

In future I'd like to have it create a tag for the version if:

a) no such tag already exists
b) the compiletest doesn't have errors
c) it's on the main branch

It might then be able to trigger CI's for our different T2 distros

we might want to make the version have a -x at the end e.g. 6.1.2-1 so
we can release fixes between versions

### CODE REVIEW

**Comment by AdityaGarg8** [comment]
We probably can exclude bcm5974 and applesmc from checkpatch since they shall not be sent upstream

File: .github/workflows/compile_test.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - name: Install Build Dependencies
+      run: |
+        sudo apt-get install -y build-essential fakeroot libncurses-dev bison flex libssl-dev libelf-dev \
+            openssl dkms libudev-dev libpci-dev libiberty-dev autoconf wget xz-utils git \
+            libcap-dev bc rsync cpio dh-modaliases debhelper kernel-wedge curl gawk dwarves zstd
+    - name: Checkpatch
+      run: |
+        KVER=$(cat version)
+        cd linux-$KVER
+        # Skip apple-bce because it needs a lot of formatting changes, and
```
Comment (AdityaGarg8) [inline .github/workflows/compile_test.yml:30]: We probably can exclude bcm5974 and applesmc from checkpatch since they shall not be sent upstream

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]

**Review by AdityaGarg8 — APPROVED** [review]

**Comment by AdityaGarg8** [comment]
Btw over here do we intend to update the kernel version in this repo and push the same to all the kernel repos?

**Comment by AdityaGarg8** [comment]
Use an action similar to https://github.com/marketplace/actions/push-directory-to-another-repository

**Comment by Redecorating** [comment]
> Btw over here do we intend to update the kernel version in this repo and push the same to all the kernel repos?

we could do that in the future. For arch at least i've been waiting until after the official arch version of a kernel is out before releasing a t2 arch kernel, so that probably would change if we automate it as you've described.

**Comment by AdityaGarg8** [comment]
> we could do that in the future. For arch at least i've been waiting until after the official arch version of a kernel is out before releasing a t2 arch kernel, so that probably would change if we automate it as you've described.

Fair enough. I've been using a cron job which watches kernel.org and updates the repo automatically for Ubuntu.

**Pull request merged**